### PR TITLE
SP2-575: Adds cypress tests for Adding Steps

### PR DESCRIPTION
**Pull request by pipwilson**

### CODE REVIEW

File: integration_tests/e2e/add-steps.cy.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+describe('Add Steps', () => {
+  const addStep = new AddSteps()
+
+  const selectStepDescriptionByIndex = (index: number) => {
+    return cy.get(`table.goal-summary-card__steps .govuk-table__body > :nth-child(${index}) > :nth-child(1)`)
+  }
+
+  const selectStepActorByIndex = (index: number) => {
+    return cy.get(`table.goal-summary-card__steps .govuk-table__body > :nth-child(${index}) > :nth-child(2)`)
+  }
```
Comment (Thomas-Geraghty) [inline integration_tests/e2e/add-steps.cy.ts]: Is it possible to move these out of the `describe('Add steps,()...)` function? Still in this file, just not defining the function within the test itself.

File: integration_tests/e2e/add-steps.cy.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        })
+      })
+    })
+
+    it('Add one step', () => {
+      cy.url().should('include', '/add-steps')
+      const stepDescription = 'This is the first step'
+      const stepActor = 'Sam'
+      addStep.addStepAutocompleteText(1, stepDescription)
+      addStep.selectStepActor(1, stepActor)
```
Comment (Thomas-Geraghty) [inline integration_tests/e2e/add-steps.cy.ts]: Any reason you didn't use the DataGenerator.generateStep() for generating some of this info? could have done
```
const step = DataGenerator.generateStep()

addStep.addStepAutocompleteText(1, step.description)
addStep.selectStepActor(1, step.actor)
```

No real difference in the end result, but it just means that, like in later steps, if you're creating multiple steps you don't need to make multiple step related variables

Comment (pipwilson) [reply integration_tests/e2e/add-steps.cy.ts]: because I forgot it existed. I will do that, much better.

Comment (pipwilson) [reply integration_tests/e2e/add-steps.cy.ts]: Addressed in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/124/commits/3e866574dd4374e959a1bf04f0401f6c8616474a

File: integration_tests/e2e/add-steps.cy.ts (hunk 1 above)
Comment (pipwilson) [reply integration_tests/e2e/add-steps.cy.ts]: Addressed in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/124/commits/3e866574dd4374e959a1bf04f0401f6c8616474a

**Review by Thomas-Geraghty — APPROVED** [review]

**Pull request merged**